WalletWasabi.Fluent/Models/Wallets/WalletModel.cs

```diff
@@ -73,8 +74,57 @@ public WalletModel(Wallet wallet, IAmountProvider amountProvider)
 
 	internal Wallet Wallet { get; }
 
-	public string Name => Wallet.WalletName;
+	public int Id => Wallet.Id;
 
+	public string Name
+	{
+		get => Wallet.WalletName;
+		set
+		{
+			if (value == null)
+			{
+				throw new InvalidOperationException("Wallet name can't be set to null");
+			}
+
+			if (!IsValidWalletName(value))
+			{
+				Logger.LogWarning($"Invalid name '{value}' when attempting to rename '{Wallet.WalletName}'");
+				throw new InvalidOperationException($"Invalid name {value}");
+			}
+
+			string walletDir = Services.WalletManager.WalletDirectories.WalletsDir;
+			var oldWalletPath = Path.Combine(walletDir, Wallet.WalletName + "." + WalletDirectories.WalletFileExtension);
+			var newWalletPath = Path.Combine(walletDir, value + "." + WalletDirectories.WalletFileExtension);
+			Rename(oldWalletPath, newWalletPath);
+			try
+			{
+				string backupDir = Services.WalletManager.WalletDirectories.WalletsBackupDir;
+				var oldBackupPath = Path.Combine(backupDir, Wallet.WalletName + "." + WalletDirectories.WalletFileExtension);
+				var newBackupPath = Path.Combine(backupDir, value + "." + WalletDirectories.WalletFileExtension);
+				Rename(oldBackupPath, newBackupPath);
+			}
+			catch (Exception e)
+			{
+				Logger.LogWarning($"Could not rename wallet backup file. Reason: {e.Message}");
+			}
+
+			Wallet.SetFilePath(newWalletPath);
+			
+			this.RaisePropertyChanged();
+		}
+	}
```

Comment:
Well, the information expert (principle from [GRASP](https://en.wikipedia.org/wiki/GRASP_(object-oriented_design))) in this case is the **Wallet**. Nobody knows better what names a **Wallet** should take than the **Wallet** itself. 

Moving the logic out of the model would go towards an **anemic model** (without behavior), and that's problematic because it reduces de expresiveness of the model. It would turn **WalletModel** into a DTO, depending on "services" / other layers to perform its business logic. The **ViewModel** should act as a glue for other pieces in the model to be "displayable" for each use case, but not perform the actual business logic.

For instance, imagine that someone wants to create a test for this logic. It feels wrong to test WalletSettingsViewModel to verify that wallets can be renamed. 

As a side note, the fact that wallet names deal with files isn't relevant because it's only an implementation detail. This doesn't make the business logic go out of the **Wallet**.